### PR TITLE
feat: #188 add test icon

### PR DESCRIPTION
**Pull request by Alex89198900**
1. Add test icon

### CODE REVIEW

File: mw-webapp/src/component/icon/Icon.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      size={props.size}
+      className={styles.testIcon}
+    />
+  );
+};
+
+describe("Icon component", () => {
+
+  it("shoud be renders when EyeOpenedIcon", () => {
+    cy.mount(createTestIcon({name: IconSizeIconDictionaryTest.EyeOpenedIcon, size: IconSize.MEDIUM}));
```
[the remark this answers is not in the document]
Comment (Alex89198900) [reply mw-webapp/src/component/icon/Icon.cy.tsx]: Done

File: mw-webapp/src/component/icon/Icon.module.scss [hunk 2]
```diff
@@ -13,3 +13,7 @@
   width: $iconSizeMedium;
   height: $iconSizeMedium;
 }
+
+.testIcon {
```
[the remark this answers is not in the document]
Comment (Alex89198900) [reply mw-webapp/src/component/icon/Icon.module.scss]: Done

**Comment by tritonJS826** [comment]
let's open it to MW-622/add-new-service

File: mw-webapp/src/component/icon/Icon.cy.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+};
+
+describe("Icon component", () => {
+
+  it("shoud be render", () => {
+    cy.mount(createTestIcon({name: ICON_RENDER, size: IconSize.MEDIUM}));
+    cy.get(getDataCy(ICON_CY)).should("exist");
+  });
+
+  it("shoud be render medium size", () => {
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/icon/Icon.cy.tsx:43]:
```suggestion
  it("should be render medium size", () => {
```

File: mw-webapp/src/component/icon/Icon.cy.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    cy.mount(createTestIcon({name: ICON_RENDER, size: IconSize.MEDIUM}));
+    cy.get(getDataCy(ICON_CY)) .invoke("css", "width").should("match", /25.*px/);
+    cy.get(getDataCy(ICON_CY)) .invoke("css", "height").should("match", /25.*px/);
+  });
+  it("shoud be render small size", () => {
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/icon/Icon.cy.tsx:48]:
```suggestion
  it("should be render small size", () => {
```

Comment (Ekaterina1994) [inline mw-webapp/src/component/icon/Icon.cy.tsx:48]: I think you can check just one size but not by match width with size in px but  in the way: if size small then element has styles .small and similar)

File: mw-webapp/src/component/icon/Icon.cy.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  it("shoud be render", () => {
+    cy.mount(createTestIcon({name: ICON_RENDER, size: IconSize.MEDIUM}));
+    cy.get(getDataCy(ICON_CY)).should("exist");
+  });
+
+  it("shoud be render medium size", () => {
+    cy.mount(createTestIcon({name: ICON_RENDER, size: IconSize.MEDIUM}));
+    cy.get(getDataCy(ICON_CY)) .invoke("css", "width").should("match", /25.*px/);
+    cy.get(getDataCy(ICON_CY)) .invoke("css", "height").should("match", /25.*px/);
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/icon/Icon.cy.tsx:46]:
```suggestion
    cy.get(getDataCy(ICON_CY)).invoke("prop", "size").should("eq", IconSize.MEDIUM).should("have.class", IconSize.MEDIUM);
```

File: mw-webapp/src/component/icon/Icon.cy.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import {Icon, IconDictionary, IconSize} from "src/component/icon/Icon";
+import {getDataCy} from "src/utils/cyTesting/getDataCy";
+
+const ICON_CY = "icon";
+const ICON_RENDER = "EyeOpenedIcon";
```
Comment (tritonJS826) [inline mw-webapp/src/component/icon/Icon.cy.tsx:5]: why render? it looks like just icon name

File: mw-webapp/src/component/icon/Icon.cy.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+describe("Icon component", () => {
+
+  it("shoud be render", () => {
+    cy.mount(createTestIcon({name: ICON_RENDER, size: IconSize.MEDIUM}));
+    cy.get(getDataCy(ICON_CY)).should("exist");
+  });
+
+  it("shoud be render medium size", () => {
+    cy.mount(createTestIcon({name: ICON_RENDER, size: IconSize.MEDIUM}));
+    cy.get(getDataCy(ICON_CY)) .invoke("css", "width").should("match", /25.*px/);
```
Comment (tritonJS826) [inline mw-webapp/src/component/icon/Icon.cy.tsx:45]: lets check the classname too in the same test

File: mw-webapp/src/component/icon/Icon.cy.tsx (hunk 4 above)
Comment (tritonJS826) [reply mw-webapp/src/component/icon/Icon.cy.tsx:48]: we can check all sizes it is different classname, dictionary with the classname could be broken one time

File: mw-webapp/src/component/icon/Icon.cy.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    cy.get(getDataCy(ICON_CY)) .invoke("css", "height").should("match", /25.*px/);
+  });
+  it("shoud be render small size", () => {
+    cy.mount(createTestIcon({name: ICON_RENDER, size: IconSize.SMALL}));
+    cy.get(getDataCy(ICON_CY)) .invoke("css", "width").should("match", /15.*px/);
+    cy.get(getDataCy(ICON_CY)) .invoke("css", "height").should("match", /15.*px/);
```
Comment (tritonJS826) [inline mw-webapp/src/component/icon/Icon.cy.tsx:51]: what do you think about shorter syntax:

```suggestion
    cy.get(getDataCy(ICON_CY)) .invoke("css", "width").should("match", /15.*px/).should("match", /15.*px/);
```

File: mw-webapp/src/component/icon/Icon.cy.tsx (hunk 3 above)
Comment (Alex89198900) [reply mw-webapp/src/component/icon/Icon.cy.tsx:43]: Done

File: mw-webapp/src/component/icon/Icon.cy.tsx (hunk 4 above)
Comment (Alex89198900) [reply mw-webapp/src/component/icon/Icon.cy.tsx:48]: Done

Comment (Alex89198900) [reply mw-webapp/src/component/icon/Icon.cy.tsx]: Done

File: mw-webapp/src/component/icon/Icon.cy.tsx (hunk 5 above)
Comment (Alex89198900) [reply mw-webapp/src/component/icon/Icon.cy.tsx]: Done

File: mw-webapp/src/component/icon/Icon.cy.tsx (hunk 6 above)
Comment (Alex89198900) [reply mw-webapp/src/component/icon/Icon.cy.tsx]: Done

File: mw-webapp/src/component/icon/Icon.cy.tsx (hunk 7 above)
Comment (Alex89198900) [reply mw-webapp/src/component/icon/Icon.cy.tsx]: Done

File: mw-webapp/src/component/icon/Icon.cy.tsx (hunk 8 above)
Comment (Alex89198900) [reply mw-webapp/src/component/icon/Icon.cy.tsx]: Done

**Review by Ekaterina1994 — APPROVED** [review]

File: mw-webapp/src/component/icon/Icon.cy.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  });
+
+  it("should be render medium size", () => {
+    cy.mount(createTestIcon({name: ICON, size: IconSize.MEDIUM}));
+    cy.get(getDataCy(ICON_CY)).invoke("css", "width").should("match", /25.*px/).should("match", /25.*px/);
+    cy.get(getDataCy(ICON_CY)).should("have.class", styles[IconSize.MEDIUM]);
+  });
+  it("should be render small size", () => {
+    cy.mount(createTestIcon({name: ICON, size: IconSize.SMALL}));
+    cy.get(getDataCy(ICON_CY)).invoke("css", "width").should("match", /15.*px/).should("match", /15.*px/);
```
Comment (tritonJS826) [inline mw-webapp/src/component/icon/Icon.cy.tsx:51]:
```suggestion
    cy.get(getDataCy(ICON_CY)).invoke("css", "width").should("match", /15.*px/);
```

File: mw-webapp/src/component/icon/Icon.cy.tsx [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+describe("Icon component", () => {
+
+  it("should be render", () => {
+    cy.mount(createTestIcon({name: ICON, size: IconSize.MEDIUM}));
+    cy.get(getDataCy(ICON_CY)).should("exist");
+  });
+
+  it("should be render medium size", () => {
+    cy.mount(createTestIcon({name: ICON, size: IconSize.MEDIUM}));
+    cy.get(getDataCy(ICON_CY)).invoke("css", "width").should("match", /25.*px/).should("match", /25.*px/);
```
Comment (tritonJS826) [inline mw-webapp/src/component/icon/Icon.cy.tsx:46]:
```suggestion
    cy.get(getDataCy(ICON_CY)).invoke("css", "width").should("match", /25.*px/);
```

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**